### PR TITLE
Version 2.1.4.231227

### PR DESCRIPTION
**Pull request by SC426**
pandas-stubs 2.1.4.231227

**Destination channel:** defaults

### Links

- [PKG-3737](https://anaconda.atlassian.net/browse/PKG-3737) 
- [Upstream repository](https://github.com/pandas-dev/pandas-stubs)

### Explanation of changes:

- Bump version to 2.1.4
- Update build skip to `py<39`
- Update dependencies
- skip `missing_imports_or_run_test_py`

[PKG-3737]: https://anaconda.atlassian.net/browse/PKG-3737?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,6 +25,7 @@ requirements:
   run:
     - python
     - types-pytz >=2022.1.1
+    - {{ pin_compatible('numpy', lower_bound='1.26.0')}}
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - numpy >=1.26 # [py<313]
```
pin_compatible is to ensure consistency with the version used in host. When only the python numpy api is used, we don't need to set it in host and therefore don't need pin_compatible.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -27,6 +25,7 @@ requirements:
   run:
```
Comment (cbouss) [inline recipe/meta.yaml:24]: setuptools and wheel can be removed, and poetry replaced by poetry-core

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml]: That's what I get for trying to be fancy!

**Review by cbouss — APPROVED** [review]

**Review by ViridianMelody — APPROVED** [review]
LGTM!

**Pull request merged**